test/unit/bigip/test_bigip_vlan.py

```diff
@@ -0,0 +1,523 @@
+from ansible.module_utils import basic
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils.f5_utils import (
+    AnsibleF5Client,
+    F5ModuleError
+)
+
+from library.bigip_vlan import (
+    Parameters,
+    ModuleManager,
+    ArgumentSpec,
+)
+
+
+fixture_path = os.path.join(os.path.dirname(__file__), 'fixtures')
+fixture_data = {}
+
+
+def set_module_args(args):
+    args = json.dumps({'ANSIBLE_MODULE_ARGS': args})
+    basic._ANSIBLE_ARGS = to_bytes(args)
+
+
+def load_fixture(name):
+    path = os.path.join(fixture_path, name)
+
+    if path in fixture_data:
+        return fixture_data[path]
+
+    with open(path) as f:
+        data = f.read()
+
+    try:
+        data = json.loads(data)
+    except:
+        pass
+
+    fixture_data[path] = data
+    return data
+
+
+class BigIpObj(object):
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+
+class TestParameters(unittest.TestCase):
+
+    def setUp(self):
+        self.loaded_ifcs = []
+        ifcs_json = load_fixture('load_net_interfaces.json')
+        for item in ifcs_json:
+            self.loaded_ifcs.append(BigIpObj(**item))
+
+    def test_module_parameters(self):
+        args = dict(
+            name='somevlan',
+            tag=213,
+            description='fakevlan',
+            untagged_interfaces=['1.1'],
+        )
+        with patch.object(Parameters, '_get_interfaces_from_device') as obj:
+            obj.return_value = self.loaded_ifcs
+            p = Parameters(args)
+
+        assert p.name == 'somevlan'
+        assert p.tag == 213
+        assert p.description == 'fakevlan'
+        assert p.untagged_interfaces == ['1.1']
+
+    def test_api_parameters(self):
+        args = dict(
+            name='somevlan',
+            description='fakevlan',
+            tag=213,
+            tagged_interfaces=['1.2']
+        )
+
+        with patch.object(Parameters, '_get_interfaces_from_device') as obj:
+            obj.return_value = self.loaded_ifcs
+            p = Parameters(args)
+
+        assert p.name == 'somevlan'
+        assert p.tag == 213
+        assert p.interfaces == [{'tagged': True, 'name': '1.2'}]
+        assert p.description == 'fakevlan'
+
+
+class TestManager(unittest.TestCase):
+
+    def setUp(self):
+        self.spec = ArgumentSpec()
+        self.loaded_ifcs = []
+        self.loaded_vlan_ifc_tag = []
+        self.loaded_vlan_ifc_untag = []
+        ifcs_tag = load_fixture('load_vlan_tagged_ifcs.json')
+        ifcs_untag = load_fixture('load_vlan_untag_ifcs.json')
+        ifcs_json = load_fixture('load_net_interfaces.json')
+        for item in ifcs_json:
+            self.loaded_ifcs.append(BigIpObj(**item))
+        for item in ifcs_tag:
+            self.loaded_vlan_ifc_tag.append(BigIpObj(**item))
+        for item in ifcs_untag:
+            self.loaded_vlan_ifc_untag.append(BigIpObj(**item))
+
+    @patch('ansible.module_utils.f5_utils.AnsibleF5Client._get_mgmt_root',
```

Comment:
Move this to the TestManager class (decorate the class) and delete all the method decorators.